src/sfctl/tests/app_test.py

```diff
@@ -87,3 +88,81 @@ def parse_app_metrics_single_test(self):
         self.assertEqual(res.maximum_capacity, '3')
         self.assertEqual(res.reservation_capacity, '2')
         self.assertEqual(res.total_application_capacity, '2')
+
+    def parse_cluster_manifest_test(self):
+        """Parse cluster manifest loaded from a file"""
+        filepath = os.path.normpath(
+            os.path.join(
+                os.path.dirname(
+                    os.path.realpath(__file__)),
+                'clustermanifest1.txt'))
+        f = open(filepath, 'r')
+        content = f.read()
+        f.close()
+        self.assertEqual(sf_c.parse_imagestore_connection_string(content),
+                         'fabric:ImageStore')
+
+    def parse_empty_cluster_manifest_test(self): #pylint: disable=invalid-name
+        """Parse empty cluster manifest content"""
+        self.assertEqual(sf_c.parse_imagestore_connection_string(None),
+                         None)
+
+    def parse_missing_imagestore_entry_cluster_manifest_test(self): #pylint: disable=invalid-name
+        """Parse missing image store connection string in cluster manifest"""
+        filepath = os.path.normpath(
+            os.path.join(
+                os.path.dirname(
+                    os.path.realpath(__file__)),
+                'invalid_clustermanifest1.txt'))
+        f = open(filepath, 'r')
+        content = f.read()
+        f.close()
+        self.assertEqual(sf_c.parse_imagestore_connection_string(content),
+                         None)
+
+    def parse_invalid_cluster_manifest_test(self): #pylint: disable=invalid-name
+        """Parse invalid cluster manifest loaded from a file"""
+        filepath = os.path.normpath(
+            os.path.join(
+                os.path.dirname(
+                    os.path.realpath(__file__)),
+                'invalid_clustermanifest2.txt'))
+        f = open(filepath, 'r')
+        content = f.read()
+        f.close()
+        self.assertEqual(sf_c.parse_imagestore_connection_string(content),
+                         None)
+
+    def parse_fileshare_path_test(self):
+        """Parse fileshare path from the cluster manifest"""
+        filepath = os.path.normpath(
+            os.path.join(
+                os.path.dirname(
+                    os.path.realpath(__file__)),
+                'clustermanifest2.txt'))
+        f = open(filepath, 'r')
+        content = f.read()
+        f.close()
+        self.assertEqual(sf_c.parse_imagestore_connection_string(content),
+                         'file:C:\\temp')
+        self.assertEqual(sf_c.parse_file_share_path(
+            sf_c.parse_imagestore_connection_string(content)), 'C:\\temp')
+
+    def upload_to_fileshare_test(self):
+        """Uploads the package to the fileshare"""
+        import tempfile
+        import shutil
+        try:
+            src_dir = os.path.dirname(os.path.realpath(__file__))
```

Comment:
Rather than using file as a test, it's good to try and create a temporary file as well. That way all the copying occurs between tempfiles

Reply:
Be sure to also get CI passing, otherwise the submission will be blocked